gto/index.py

```diff
@@ -243,9 +245,17 @@ def get_history(self) -> Dict[str, Index]:
 ArtifactsCommits = Dict[str, ArtifactCommits]
 
 
-class RepoIndexManager(FileIndexManager):
+class RepoIndexManager(FileIndexManager, CommitChangesDueToAddMixin):
     repo: git.Repo
 
+    def __init__(self, repo, config):
+        super().__init__(repo=repo, config=config)
+        # @aguschin
```

Comment:
@aguschin Can you please help me out with this?

I was trying to follow same strategy, now for the committing part. I created the class `git_utils.CommitChangesDueToAddMixin`, and I need to store the repo somewhere in `self`. 
Why is it failing here? I suppose it must have something to do with what happens to the parent class, but it gets too complicated. Any hint? Alternatively, can I get access to `repo` from `self` in any other way?

Reply:
Sorry, missed your comment... 🤦🏻‍♂️ 
this is something with pydantic: `RepoIndexManager` is a subclass of `FileIndexManager`, which is a subclass of `BaseIndexManager`, that is a subclass of pydantic `BaseModel`.
Pydantic require some special work with fields starting with underscore https://pydantic-docs.helpmanual.io/usage/models/#private-model-attributes

Reply:
just checked, using `PrivateAttr()` (as in that page) works, but the test fails after that, though I believe it's unrelated issue.